gtdynamics/dynamics/Chain.cpp

```diff
@@ -0,0 +1,116 @@
+/* ----------------------------------------------------------------------------
+ * GTDynamics Copyright 2021, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file Chain.cpp
+ * @brief Create a serial kinematic chain.
+ * @author Dan Barladeanu, Frank Dellaert.
+ */
+
+#include <gtdynamics/dynamics/Chain.h>
+
+namespace gtdynamics {
+
+void Chain::monoidOperation(const Pose3 &aTb, const Matrix &bAj,
+                            const Pose3 &bTc, const Matrix &cAk, 
+                            Pose3 &aTc, Matrix &cAjk) {
+    
+    // expect 6 rows in jacobians
+    if (bAj.rows() != 6 || cAk.rows() != 6) {
+      throw std::runtime_error(
+          "Jacobians should have 6 rows in SE(3)");
+    }
+
+    // Compose poses:
+    aTc = aTb.compose(bTc);
```

Comment:
get c_Ad_b from compose by passing it as OptionalJacobian argument

Reply:
changed this function so the arguments are different now